worksheets/espresso-testing/index.md

```diff
@@ -0,0 +1,240 @@
+---
+layout: page
+title: Espresso UI Testing
+---
+
+## Introduction
+This tutorial follows on from the [first one where you built a cookie clicker]({{ site.baseurl }}/worksheets/1-introduction/). If you worked on that tutorial already, follow along using the code you wrote previously.
+
+If you didn't work on that tutorial, or you don't have the code anymore, you can download a completed project to follow along with [here](downloads/cookie-clicker.zip). Download the file, unzip it and then open it with Android Studio using the *Open an existing Android Studio project* button on the welcome screen.
+
+## What is UI testing?
+To make sure your software works properly you need to test it whenever you make a change that you want to release to your users. For something like an app, you need to go through all the features of the app and make sure that nothing is broken. For even a small app this is a boring, repetitive, and time-consuming job. That's precisely the sort of thing which we use computers for! Automated tests are pieces of code which test that your software is working correctly for you.
```

Comment:
Maybe we could be more positive? Otherwise students won't see why writing tests is good! Writing good UI tests has a lot of really great benefits, like confidence in new features not breaking old features etc. You can make them run regularly, on many devices, many times?